Core/src/org/sleuthkit/autopsy/threadutils/TaskRetryUtil.java

```diff
@@ -0,0 +1,280 @@
+         * a delay nor a time out.
+         *
+         * @param delay         The delay before the task should be attempted,
+         *                      may be zero or any positive integer.
+         * @param delayTimeUnit The time unit for the delay before the task
+         *                      should be attempted.
+         */
+        public TaskAttempt() {
+            this.delay = 0L;
+            this.timeOut = 0L;
+            this.timeUnit = TimeUnit.SECONDS;
+        }
+
+        /**
+         * Constructs an object that encapsulates the specification of a task
+         * attempt for the attemptTask() utility.
+         *
+         * @param delay    The delay before the task should be attempted, may be
+         *                 zero or any positive integer.
+         * @param timeUnit The time unit for the delay before the task should be
+         *                 attempted.
+         */
+        public TaskAttempt(Long delay, TimeUnit timeUnit) {
+            if (delay == null || delay < 0) {
+                throw new IllegalArgumentException(String.format("Argument for delay parameter = %d, must be zero or any positive integer", delay));
+            }
+            if (timeUnit == null) {
+                throw new IllegalArgumentException("Argument for timeUnit parameter is null");
+            }
+            this.delay = delay;
+            this.timeOut = 0L;
+            this.timeUnit = TimeUnit.SECONDS;
+        }
+
+        /**
+         * Constructs an object that encapsulates the specification of a task
+         * attempt for the attemptTask() utility.
+         *
+         * @param delay    The delay before the task should be attempted, must
+         *                 be zero or any positive integer.
+         * @param timeOut  The timeout for the task attempt, must be zero or any
+         *                 positive integer.
+         * @param timeUnit The time unit for the delay before the task should be
+         *                 attempted and the time out.
+         */
+        public TaskAttempt(Long delay, Long timeOut, TimeUnit timeUnit) {
+            if (delay == null || delay < 0) {
+                throw new IllegalArgumentException(String.format("Argument for delay parameter = %d, must be zero or any positive integer", delay));
+            }
+            if (timeOut == null || timeOut < 0) {
+                throw new IllegalArgumentException(String.format("Argument for timeOut parameter = %d, must be zero or any positive integer", delay));
+            }
+            if (timeUnit == null) {
+                throw new IllegalArgumentException("Argument for timeUnit parameter is null");
+            }
+            this.delay = delay;
+            this.timeOut = timeOut;
+            this.timeUnit = timeUnit;
+        }
+
+        /**
+         * Gets the optional delay before the task should be attempted, may be
+         * zero.
+         *
```

Comment:
I think I would note here that the units for the delay (and the timeout in the next getter) come from getTimeUnit()